design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,254 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**                                      | **Description**                                              |
+| --------------------------------------------- | ------------------------------------------------------------ |
+| Epic                                          | https://github.com/cyberark/conjur/issues/1892               |
+| Ruby CLI                                      | https://github.com/cyberark/conjur-cli                       |
+| Go CLI                                        | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI                                    | https://github.com/cyberark/conjur-api-python3               |
+| [FPM](https://github.com/jordansissel/fpm)    | For building packages for multiple platforms (RPM, DEB, OS X) |
+| [Debify](https://github.com/conjurinc/debify) | Build a package from a Ruby gem                              |
```

Comment:
are you aware of [Goreleaser's tooling](https://carlosbecker.com/posts/nfpm/) for releasing deb/rpm as part of a Golang project's pipeline? might be worth referencing here too ([secretless example](https://github.com/cyberark/secretless-broker/blob/master/.goreleaser.yml#L126-L127))

Reply:
@sigalsax and I also found https://github.com/pmq20/ruby-packer today

Reply:
@izgeri if Go allows to run a single binary that can be shipped to users (after compiled down to each target platform) why would we need to deliver them as packages (using the Goreleaser)?

Reply:
go has a separate binary for each OS, it's not a single binary.

it depends on what we want to make available. I think reviewing what standard CLIs have available as install options and deciding what we want to provide is likely the best first step.